screen/loans/src/main/java/com/ivy/loans/loan/LoanViewModel.kt

```diff
@@ -300,18 +305,23 @@ class LoanViewModel @Inject constructor(
         } ?: defaultCurrencyCode
     }
 
-    private suspend fun calculateAmountPaid(loan: Loan): Double {
+    /**
+     *  Calculates the total amount paid and the total loan amount including any changes made to the loan.
+     *  @return A Pair containing the total amount paid and the total loan amount.
+     */
+    private suspend fun calculateAmountPaidAndTotalAmount(loan: Loan): Pair<Double, Double> {
         val loanRecords = ioThread { loanRecordDao.findAllByLoanId(loanId = loan.id) }
-        var amount = 0.0
-
-        loanRecords.forEach { loanRecord ->
-            if (!loanRecord.interest) {
-                val convertedAmount = loanRecord.convertedAmount ?: loanRecord.amount
-                amount += convertedAmount
-            }
+        val (amountPaid, loanTotalAmount) = loanRecords.fold(0.0 to loan.amount) { value, loanRecord ->
+            val (currentAmountPaid, currentLoanTotalAmount) = value
+            if (loanRecord.interest) return@fold value
+            val convertedAmount = loanRecord.convertedAmount ?: loanRecord.amount
+
+            loanRecord.loanRecordType.processByType(
+                decreaseAction = { currentAmountPaid + convertedAmount to currentLoanTotalAmount },
+                increaseAction = { currentAmountPaid to currentLoanTotalAmount + convertedAmount }
+            )
         }
-
-        return amount
+        return amountPaid to loanTotalAmount
```

Comment:
nit: You can directly `return fold(..){...}` no need for destructing and reconstructing again